modules/database/src/std/link/lnkConst.c

```diff
@@ -496,7 +496,7 @@ static long lnkConst_loadArray(struct link *plink, short dbrType, void *pbuffer,
         else {
             /* Long string conversion */
             strncpy(pbuffer, clink->value.scalar_string, *pnReq);
-            ((char *)pbuffer)[*pnReq] = 0;
+            ((char *)pbuffer)[*pnReq - 1] = 0;
```

Comment:
Core group: Check for the `*pnReq == 0` case?

Reply:
So `pnReq` is the size of the buffer you are trying to write into. For something like an `aSub` record (the case where we saw this issue), the initialisation forces that the buffer has size at least one, so this cannot happen.

However, I see no reason why a custom record would have to behave that way; I could in theory have a custom record with an input link whose buffer size can be zero as well. What is the correct behaviour in that case? My inclination is that we should drop and out set `status` to 1, but I am not confident about that.